Content.Server/Power/EntitySystems/PowerReceiverSystem.cs

```diff
@@ -160,7 +173,33 @@ public bool IsPowered(EntityUid uid, ApcPowerReceiverComponent? receiver = null)
             if (!_recQuery.Resolve(uid, ref receiver, false))
                 return true;
 
-            return receiver.Powered;
+            return !receiver.PowerDisabled
+                   && (!receiver.NeedsPower
+                       || MathHelper.CloseToPercent(receiver.NetworkLoad.ReceivingPower, receiver.Load));
+        }
+
+        public void UpdateIsPowered(Entity<ApcPowerReceiverComponent?, MetaDataComponent?> entity)
+        {
+            SetIsPowered(entity, IsPowered(entity.Owner, entity.Comp1));
+        }
+
+        private void SetIsPowered(Entity<ApcPowerReceiverComponent?, MetaDataComponent?> entity, bool powered)
```

Comment:
prob better to merge these two and get rid of nullability and resolves (this is the kind of method the caller should give the components by themselves, also metadata is always safe to get with Comp)

Reply:
Done